### PR TITLE
JWT Token Authentication

### PR DESCRIPTION
**Pull request by driusan**
This adds support for the final piece of v0.0.1 of the LORIS API, user authentication with JWT tokens. Users can now post to login in the API with a valid username and password to receive a token which can be used for authentication in an Authorization: HTTP header.

The login page is also updated to use the API's authentication via javascript, so that we're using our own API.

### CODE REVIEW

**Comment by jstirling91** [comment]
Get a ``` 500 Internal Server Error``` when posting to /login. The error log says 

```Config setting JWTKey does not exist in database```

Should this setting be added to the XML or the Database?

**Comment by samirdas** [comment]
Put a release note for this pull request if it still fails PHP 5.4. Watch this!

File: SQL/0000-00-03-ConfigTables.sql [hunk 1]
```diff
@@ -166,3 +170,5 @@ INSERT INTO Config (ConfigID, Value) SELECT ID, "mri_parameter_form" FROM Config
 INSERT INTO Config (ConfigID, Value) SELECT ID, "no-reply@example.com" FROM ConfigSettings WHERE Name="From";
 INSERT INTO Config (ConfigID, Value) SELECT ID, "no-reply@example.com" FROM ConfigSettings WHERE Name="Reply-to";
 INSERT INTO Config (ConfigID, Value) SELECT ID, "Produced by LorisDB" FROM ConfigSettings WHERE Name="X-MimeOLE";
+
+INSERT INTO Config (ConfigID, Value) SELECT ID, "S3cret" FROM ConfigSettings WHERE Name="JWTKey";
```
Comment (samirdas) [inline SQL/0000-00-03-ConfigTables.sql]: Can we make a note in the documentation not in the installation for developers to change.

**Comment by christinerogers** [comment]
This pull request means you have run composer update

**Pull request merged**